base/src/types.rs

```diff
@@ -106,8 +106,13 @@ pub enum Type<Id, T = AstType<Id>> {
     Record {
         /// The associated types of this record type
         types: Vec<Field<Id, Alias<Id, T>>>,
+        row: T,
+    },
+    EmptyRow,
```

Comment:
Oh I see - `Record` has a `row` field now. I recon call these `EmptyRecord` and `ExtendRecord`